crates/bevy_pbr/src/render/pbr.wgsl

```diff
@@ -18,13 +19,28 @@ struct FragmentInput {
 
 @fragment
 fn fragment(in: FragmentInput) -> @location(0) vec4<f32> {
+    let is_orthographic = view.projection[3].w == 1.0;
+    let V = calculate_view(in.world_position, is_orthographic);
+    var uv = in.uv;
+#ifdef VERTEX_UVS
+#ifdef VERTEX_TANGENTS
+    if ((material.flags & STANDARD_MATERIAL_FLAGS_DEPTH_MAP_BIT) != 0u) {
+        let tangent_V = vec3<f32>(
+            dot(V, in.world_tangent.xyz),
+            dot(V, -cross(in.world_normal, in.world_tangent.xyz) * sign(in.world_tangent.w)),
+            dot(V, in.world_normal),
+        );
+        uv = parallaxed_uv(material.parallax_depth, material.max_parallax_layer_count, uv, tangent_V);
```

Comment:
In our normal-mapping code (`apply_normal_mapping()`) we use this:
```rust
    var N: vec3<f32> = world_normal;
    var T: vec3<f32> = world_tangent.xyz;
    var B: vec3<f32> = world_tangent.w * cross(N, T);

... stuff to calculate what Nt, the tangent-space normal sampled from the normal map should be...

    N = Nt.x * T + Nt.y * B + Nt.z * N;
```
...but that is transforming from tangent space to world space.

It took me a while to identify that the normal mapping code can be reformulated as:
```rust
N = Nt.x * T + Nt.y * B + Nt.z * N;

// Focusing on the x component of N
N.x = Nt.x * T.x + Nt.y * B.x + Nt.z * N.x;
// Which is:
N.x = dot(Nt, vec3(T.x, B.x, N.x));
```

At that point I realised from your code with:
```rust
V.x = dot(V, T);
```
that the normal mapping code is using the forward transform, taking the rows of the TBN matrix and multiplying with the tangent-space normal to transform it to world space. And that to transform the view vector from world space to tangent space would need the inverse. I remembered that the inverse of an orthonormal matrix is its transpose, which explains the difference between the two bits of code.

However, there is a minus sign in your code that looks wrong to me. It should be:
```suggestion
        // Transform the view vector from world space to tangent space by
        // multiplying it by the inverse of the TBN matrix. Given the TBN
        // matrix is orthonormal, its inverse is its transpose.
        let Vt = vec3<f32>(
            dot(V, in.world_tangent.xyz),
            dot(V, in.world_tangent.w * cross(in.world_normal, in.world_tangent.xyz),
            dot(V, in.world_normal),
        );
        uv = parallaxed_uv(material.parallax_depth, material.max_parallax_layer_count, uv, Vt);
```

If there are problems with signs after that, it could be the case that the normal maps being used have their y flipped (authored for DirectX for example) and that should be solved separately.

Reply:
The normal map is **not** applied to `world_normal` here though. You can convince yourself of it by setting `normal_map_texture` at line 256 of the `parallax_mapping.rs` example to `None`. You'll notice that it doesn't affect the parallaxing in any way.

Reply:
Sorry, that last sentence I wrote was incorrect. I meant to write if there are problems with signs after that, it could be that the depth maps being used have their y flipped and that should be solved separately. It would seem odd to me if a normal map texture was y-flipped but a depth map was not. The reason the normal map is y-flipped from our perspective of right-handed x-right, y-up. z-back is because of a different handedness convention and I would think normal and depth maps are authored together.

Reply:
Removing the minus sign does indeed break parallaxing.